app/models/user.rb

```diff
@@ -219,7 +219,10 @@ class User < ApplicationRecord
   validates :unspent_credits_count, presence: true
   validates :username, length: { in: 2..USERNAME_MAX_LENGTH }, format: USERNAME_REGEXP
   validates :username, presence: true, exclusion: { in: ReservedWords.all, message: MESSAGES[:invalid_username] }
-  validates :username, uniqueness: { case_sensitive: false }, if: :username_changed?
+  validates :username, uniqueness: { case_sensitive: false, message: ->(_obj, data) do
+        "<a href='/#{data[:value]}'>#{data[:value]}</a> is taken."
+      end
+    }, if: :username_changed?
```

Comment:
I noticed this is formatted oddly, could you run `bundle exec rubocop -a app/models/user.rb` locally?

Aside from formatting I think we should  add `target="_blank" rel="noopener"` to the `<a>` tag so that it opens in a separate tab

cc @maestromac how do you feel about having HTML here?